### PR TITLE
Add nulls comparison to OrderFilter

### PR DESCRIPTION
**Pull request by teohhanhui**
Documents api-platform/core#838

### CODE REVIEW

File: core/filters.md [hunk 1]
```diff
@@ -321,10 +321,36 @@ will not be applied unless you configure a default order direction to use:
 services:
     offer.order_filter:
         parent:    'api_platform.doctrine.orm.order_filter'
-        arguments: [ { id: 'ASC', name: 'DESC' } ]
+        arguments: [ { id: { default_direction: 'ASC' }, name: { default_direction: 'DESC' } } ]
```
Comment (dunglas) [inline core/filters.md]: Does the short syntax still works? It looks very verbose here.

File: core/filters.md [hunk 2]
```diff
@@ -321,10 +321,36 @@ will not be applied unless you configure a default order direction to use:
...
         tags:      [ { name: 'api_platform.filter', id: 'offer.order' } ]
 ```
 
+### Comparing with Null Values
```
Comment (dunglas) [inline core/filters.md]: `null` Values?

File: core/filters.md (hunk 1 above)
Comment (teohhanhui) [inline core/filters.md]: The shorthand syntax is retained for BC, but I'd like to see it deprecated. What do you think?

Comment (teohhanhui) [inline core/filters.md]: I think we can use `~` for one of the two properties here. It's good to show the usual / more common case where you don't specify a default direction.

File: core/filters.md (hunk 2 above)
Comment (teohhanhui) [inline core/filters.md]: I think we should keep that out of the headings and put it in plain English.

File: core/filters.md (hunk 1 above)
Comment (dunglas) [inline core/filters.md]: I prefer to keep the short syntax as is because it's the most common use case and it must be as less verbose as possible.

**Comment by teohhanhui** [comment]
I'd argue that the most common use case is actually `~` (`null`)... I think
it's much preferable to be explicit, instead of having a default direction
which is not visible.

Sometimes I feel like calling it a misfeature. :laugh:

File: core/filters.md (hunk 1 above)
Comment (teohhanhui) [inline core/filters.md]: I do not agree that it's "the most common use case".

**Comment by dunglas** [comment]
Thanks @teohhanhui
